promgen/static/js/promgen.vue.js

```diff
@@ -149,26 +149,43 @@ const app = Vue.createApp({
 
 app.config.compilerOptions.whitespace = "preserve";
 
-app.component('silence-form', {
-    template: '#silence-form-template',
+app.component('silence-modal', {
+    template: '#silence-modal-template',
     delimiters: ['[[', ']]'],
     data: () => ({
         state: silenceStore.state,
         form: {}
     }),
+    computed: {
+        globalMessages() {
+            return globalStore.state.messages;
+        },
+    },
     methods: {
+        addLabel() {
+            if (this.form.label && this.form.value) {
+                silenceStore.addLabel(this.form.label, this.form.value);
+                this.form.label = '';
+                this.form.value = '';
+            }
+        },
         removeLabel(label) {
             delete this.state.labels[label];
         },
         submit() {
             const body = JSON.stringify({
                 labels: this.state.labels,
-                ...this.form
+                startsAt: this.form.startsAt,
+                endsAt: this.form.endsAt,
+                duration: this.form.duration,
+                createdBy: this.form.createdBy,
+                comment: this.form.comment
             });
 
             fetch('/proxy/v1/silences', { method: 'POST', body })
                 .then(response => {
                     if (response.ok) {
+                        this.hideModal('silenceModal');
```

Comment:
```suggestion
```

This line is not necessary since the `location.reload();` that we have below will refresh the whole page.